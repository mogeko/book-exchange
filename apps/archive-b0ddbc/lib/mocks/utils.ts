import { faker } from "@faker-js/faker";

export function oneOf<T = any>(array: T[]): T {
  return array[randomNum(array.length - 1)];
}

export function someOf<T = any>(array: T[]): T[] {
  const result = array.filter(randomBool);

  return result.length === 0 ? [oneOf(array)] : result;
}

export function languages<T = any>(fn: (lang: string[]) => T): T {
  const langs = [
    "Afrikaans",
    "Arabic",
    "Azerbaijani",
    "Czech",
    "German",
    "Greek",
    "English",
    "Spanish",
    "Farsi",
    "Finnish",
    "French",
    "Georgian",
    "Hebrew",
    "Hrvatski",
    "Hungarian",
    "Armenian",
    "Indonesia",
    "Italian",
    "Japanese",
    "Korean",
    "Latvian",
    "Macedonian",
    "Norwegian",
    "Nepalese",
    "Dutch",
    "Polish",
    "Romanian",
    "Russian",
    "Slovakian",
    "Swedish",
    "Turkish",
    "Ukrainian",
    "Urdu",
    "Vietnamese",
    "Chinese",
  ];

  return fn(langs);
}

export const randomNum = faker.datatype.number;

export const randomBool = faker.datatype.boolean;

export const randomDateRecent = () => {
  return faker.date.recent(10, "2022-01-01T00:00:00.000Z").toISOString();
};

export const randomDatePost = () => {
  return faker.date.past(1000, "2000-01-01T00:00:00.000Z").toISOString();
};
