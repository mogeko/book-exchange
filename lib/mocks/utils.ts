import { faker } from "@faker-js/faker";

export function oneOf<T = any>(array: T[]): T {
  return array[randomNum(array.length - 1)];
}

export function someOf<T = any>(array: T[]): T[] {
  const result = array.filter(randomBool);

  return result.length === 0 ? [oneOf(array)] : result;
}

export function languages<T = any>(fn: (lang: string[]) => T): T {
  const langs = [
    "Afrikaans",
    "Arabic",
    "Azerbaijani",
    "Czech",
    "German",
    "Greek",
    "English",
    "Spanish",
    "Farsi",
    "Finnish",
    "French",
    "Georgian",
    "Hebrew",
    "Hrvatski",
    "Hungarian",
    "Armenian",
    "Indonesia",
    "Italian",
    "Japanese",
    "Korean",
    "Latvian",
    "Macedonian",
    "Norwegian",
    "Nepalese",
    "Dutch",
    "Polish",
    "Romanian",
    "Russian",
    "Slovakian",
    "Swedish",
    "Turkish",
    "Ukrainian",
    "Urdu",
    "Vietnamese",
    "Chinese",
  ];

  return fn(langs);
}

export const randomNum = faker.datatype.number;

export const randomBool = faker.datatype.boolean;
