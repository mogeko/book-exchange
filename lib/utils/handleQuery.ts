function handleQuery<T extends ParamProps>(url: URL, params: T) {
  if (Object.keys(params).length === 0) return url;
  const queryParams = Object.entries(params)
    .filter(([_, value]) => value)
    .map(([key, value]) => {
      if (Array.isArray(value)) {
        return value
          .filter(Boolean)
          .map((v) => `${key}=${v}`)
          .join("&");
      }

      return `${key}=${value}`;
    })
    .join("&");

  return [url, queryParams].join("?");
}

interface ParamProps {
  [key: string]: any;
}

export type URL = `/api/${string}`;
export type Query = [URL | undefined, ParamProps];

export default handleQuery;
