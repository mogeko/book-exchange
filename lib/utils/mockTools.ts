import { faker } from "@faker-js/faker";

export function oneOf(array: any[]) {
  return array[randomNum(array.length - 1)];
}

export function randomLanguage() {
  return oneOf([
    "Afrikaans",
    "Arabic",
    "Azerbaijani",
    "Czech",
    "German",
    "Greek",
    "English",
    "Spanish",
    "Farsi",
    "Finnish",
    "French",
    "Georgian",
    "Hebrew",
    "Hrvatski",
    "Hungarian",
    "Armenian",
    "Indonesia",
    "Italian",
    "Japanese",
    "Korean",
    "Latvian",
    "Macedonian",
    "Norwegian",
    "Nepalese",
    "Dutch",
    "Polish",
    "Romanian",
    "Russian",
    "Slovakian",
    "Swedish",
    "Turkish",
    "Ukrainian",
    "Urdu",
    "Vietnamese",
    "Chinese",
  ]);
}

export function arrayBy(length: number, value: any) {
  return Array.from({ length: length }, () => value);
}

export const listBy = arrayBy;

export const randomNum = faker.datatype.number;
