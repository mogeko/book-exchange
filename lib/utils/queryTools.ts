function handleQuery<T extends {}>(url: `/${string}`, params: T) {
  if (Object.keys(params).length === 0) return url;
  const queryParams = Object.entries(params)
    .filter(([_, value]) => value)
    .map(([key, value]) => `${key}=${value}`)
    .join("&");
  return [url, queryParams].join("?");
}

export default handleQuery;
