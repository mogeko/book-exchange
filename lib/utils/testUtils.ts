import { render, RenderOptions } from "@testing-library/react";
import { WrapProvider } from "@/pages/_app";

function renderWithProvider(ui: React.ReactElement, options?: RenderOptions) {
  return render(ui, { wrapper: WrapProvider, ...options });
}

export * from "@testing-library/react";
export { renderWithProvider as render };
