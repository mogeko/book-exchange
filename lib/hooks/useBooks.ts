import useSWRInfinite, { type SWRInfiniteResponse } from "swr/infinite";
import handleQuery, { type QueryParamType } from "@/lib/utils/queryTools";
import useSWR, { type SWRResponse } from "swr";
import { faker } from "@faker-js/faker";

function useBooks(queryParam: QueryParamProps = {}) {
  const query = handleQuery("/books", queryParam);
  const { data, error, ...otherRes }: SWRResponse<BookTypes[]> = useSWR(query);

  return {
    books: data,
    isLoading: !error && !data,
    isError: error,
    ...otherRes,
  };
}

export function useBook(id: string | number, queryParam: QueryParamProps = {}) {
  const query = handleQuery(`/books/${id}`, queryParam);
  const { data, error, ...otherRes }: SWRResponse<BookTypes> = useSWR(query);

  return {
    book: data,
    isLoading: !error && !data,
    isError: error,
    ...otherRes,
  };
}

export function useBooksInfinite(queryParam: QueryParamInfiniteProps) {
  const { offset = 0, ...other } = queryParam;
  const opts = { revalidateFirstPage: false };
  const { data, error, ...otherRes }: SWRInfiniteResponse<BookTypes[]> =
    useSWRInfinite((index, previous) => {
      if (previous && !previous.length) return null;
      return handleQuery("/books", {
        offset: index * other.limit! + offset!,
        ...other,
      });
    }, opts);

  return {
    data: data,
    isLoading: !error && !data,
    isError: error,
    ...otherRes,
  };
}

export const exampleData = {
  title: faker.word.noun(20),
  cover: faker.image.image(1280, 1910),
  description: faker.lorem.paragraph(10),
  published: faker.date.past().toISOString(),
  publisher: faker.company.companyName(),
  tags: faker.lorem.words(1),
  author: faker.name.firstName(),
  rates: faker.datatype.number(100),
  id: faker.datatype.number(100),
};

export type BookTypes = Partial<typeof exampleData>;

type QueryParamProps = QueryParamType<keyof BookTypes>;

type QueryParamInfiniteProps = {
  limit: number;
} & Omit<QueryParamProps, "limit">;

export default useBooks;
