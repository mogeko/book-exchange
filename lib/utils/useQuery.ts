import handleQuery from "@/lib/utils/queryTools";
import useSWR, { type SWRResponse, type SWRConfiguration } from "swr";
import useSWRInfinite, {
  type SWRInfiniteConfiguration,
  type SWRInfiniteResponse,
} from "swr/infinite";

function useQuery<T, U = {}>(
  url: `/${string}`,
  param: U,
  opts: SWRConfiguration<T> = {}
) {
  const query = handleQuery(url, param);
  const res: SWRResponse<T> = useSWR(query, opts);
  const { data, error, ...otherRes } = res;

  return {
    data,
    isLoading: !error && !data,
    isError: error,
    ...otherRes,
  };
}

export function useQueryInfinite<T>(
  getKey: (index: number, previous: T | null) => string | null,
  opts: SWRInfiniteConfiguration<T> = {}
) {
  const res: SWRInfiniteResponse<T> = useSWRInfinite(getKey, opts);
  const { data, error, ...otherRes } = res;

  return {
    data: data,
    isLoading: !error && !data,
    isError: error,
    ...otherRes,
  };
}

export default useQuery;
