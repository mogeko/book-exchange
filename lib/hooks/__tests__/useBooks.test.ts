import "@testing-library/jest-dom";
import { renderHook, waitFor } from "@/lib/utils/testTools";
import useBooks, { useBook, useBooksInfinite } from "@/lib/hooks/useBooks";

describe("useBooks", () => {
  it("should be defined", () => {
    expect(useBooks).toBeDefined();
  });

  it("should return books", async () => {
    const { result } = renderHook(() => useBooks());

    expect(result.current.data).toBeUndefined();

    await waitFor(() => {
      expect(result.current.data).toBeDefined();
      expect(result.current.data).toHaveLength(10);
    });

    expect(result.current).toMatchSnapshot();
  });
});

describe("useBook", () => {
  it("should be defined", () => {
    expect(useBook).toBeDefined();
  });

  it("should return book", async () => {
    const { result } = renderHook(() => useBook("bk1000"));

    expect(result.current.data).toBeUndefined();

    await waitFor(() => {
      expect(result.current.data).toBeDefined();
      expect(result.current.data?.id).toBe("bk1000");
    });

    expect(result.current).toMatchSnapshot();
  });

  it("return null when book ID is undefined", async () => {
    const { result } = renderHook(() => useBook());

    expect(result.current.data).toBeUndefined();

    await waitFor(() => expect(result.current.data).toBeUndefined());
  });
});

describe("useBooksInfinite", () => {
  it("should be defined", () => {
    expect(useBooksInfinite).toBeDefined();
  });

  it("should return books", async () => {
    const { result } = renderHook(() => useBooksInfinite());

    expect(result.current.data).toBeUndefined();

    await waitFor(() => {
      expect(result.current.data).toBeDefined();
      expect(result.current.data?.[0]).toHaveLength(10);
    });

    expect(result.current).toMatchSnapshot();
  });

  it("return books when query has param", async () => {
    const { result } = renderHook(() => useBooksInfinite({ limit: 5 }));

    await waitFor(() => {
      expect(result.current.data).toBeDefined();
      expect(result.current.data?.[0]).toHaveLength(5);
    });
  });
});
