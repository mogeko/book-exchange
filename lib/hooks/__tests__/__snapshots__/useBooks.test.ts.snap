// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useBook should return book 1`] = `
Object {
  "data": Object {
    "cover": "https://loremflickr.com/1280/1910/animals",
    "desc": Object {
      "is_folded": true,
      "text": "Rerum distinctio atque consequatur amet iure sequi voluptas dolorem. Suscipit non quam. Deserunt omnis sint similique cum aliquid est sunt odio nemo. Inventore est ut ut magnam ea dolores ipsum ea recusandae. Et facere cumque quae fuga dolor ipsa dolor. Unde sit cupiditate. Sint aspernatur ullam a est repellendus ducimus. Maxime id est dolor nobis fuga. Voluptatem veniam maiores aut laborum perferendis harum quo minus. Consequatur natus nihil dicta et. Placeat et aut earum harum a. Impedit occaecati magni fuga et cumque excepturi enim voluptas. Quasi ipsum ut.",
    },
    "digest": Object {
      "is_folded": true,
      "text": "Sequi voluptates delectus ullam laborum. Porro et consequuntur harum. Hic praesentium maiores qui asperiores qui vero accusantium nemo voluptas. Sequi voluptatum perferendis aperiam qui assumenda velit necessitatibus. Adipisci animi molestias aspernatur quas vel consequatur veniam. Nisi voluptas eos. Facere laboriosam rerum quidem possimus nisi sit totam veniam. Repellendus magni et similique consequatur vel officiis ullam. Est quidem ullam eum voluptas porro et voluptates doloribus. Aut minima reprehenderit commodi nemo ducimus atque.",
    },
    "id": "bk1000",
    "mate": Object {
      "author": "Gardner",
      "isbn": "978-1798832496",
      "language": "English",
      "paperback": 349,
      "publication_date": "2021-10-14T11:31:41.496Z",
      "publisher": "Mante Inc",
      "subtitle": "Officiis adipisci rerum dolorum cum et totam eveniet culpa qui.",
    },
    "rates": 41,
    "tags": Array [
      "deleniti quia",
      "doloremque dolorum",
      "et sed",
      "ut facilis",
      "et dolorem",
      "non omnis",
      "qui neque",
    ],
    "title": "text",
  },
  "error": undefined,
  "isLoading": false,
  "isValidating": false,
  "mutate": [Function],
}
`;

exports[`useBooks should return books 1`] = `
Object {
  "data": Array [
    Object {
      "cover": "https://loremflickr.com/1280/1910/sports",
      "desc": Object {
        "text": "Quis dolore similique accusantium officiis adipisci. Dolorum cum et totam eveniet culpa qui quibusdam ad facere. Rerum distinctio atque consequatur amet iure sequi voluptas dolorem. Suscipit non quam. Deserunt omnis sint similique cum aliquid est sunt odio nemo. Inventore est ut ut magnam ea dolores ipsum ea recusandae. Et facere cumque quae fuga dolor ipsa dolor. Unde sit cupiditate. Sint aspernatur ullam a est repellendus ducimus. Maxime id est dolor nobis fuga.",
      },
      "id": "bk90",
      "mate": Object {
        "author": "Gracie",
      },
      "rates": 78,
      "tags": Array [
        "quia doloremque",
        "dolorum et",
        "sed ut",
        "facilis et",
        "dolorem non",
      ],
      "title": "catch",
    },
    Object {
      "cover": "https://loremflickr.com/1280/1910/cats",
      "desc": Object {
        "text": "Impedit occaecati magni fuga et cumque excepturi enim voluptas. Quasi ipsum ut. Quia exercitationem sequi voluptates delectus ullam laborum. Porro et consequuntur harum. Hic praesentium maiores qui asperiores qui vero accusantium nemo voluptas. Sequi voluptatum perferendis aperiam qui assumenda velit necessitatibus. Adipisci animi molestias aspernatur quas vel consequatur veniam. Nisi voluptas eos. Facere laboriosam rerum quidem possimus nisi sit totam veniam. Repellendus magni et similique consequatur vel officiis ullam. Est quidem ullam eum voluptas porro et voluptates doloribus. Aut minima reprehenderit commodi nemo ducimus atque. Minima neque natus consequatur pariatur odit quaerat.",
      },
      "id": "bk88",
      "mate": Object {
        "author": "Lonnie",
      },
      "rates": 95,
      "tags": Array [
        "aut laborum",
        "perferendis harum",
        "quo minus",
        "quis consequatur",
        "natus nihil",
        "dicta et",
        "voluptate placeat",
        "et aut",
      ],
      "title": "cheek",
    },
    Object {
      "cover": "https://loremflickr.com/1280/1910/city",
      "desc": Object {
        "text": "Dolor ut quos odio quam nihil rerum quia assumenda. Perspiciatis exercitationem in reprehenderit. Molestiae aut quia voluptatem. Voluptatum dolore officiis facere ea hic nihil voluptatem ut. Consequuntur accusamus similique odit. Repellat culpa explicabo. In ex harum veniam. Corporis fuga saepe facilis. Ut labore similique molestias nihil. Aspernatur reprehenderit neque natus. Consequatur blanditiis odio ad blanditiis ab.",
      },
      "id": "bk41",
      "mate": Object {
        "author": "Raegan",
      },
      "rates": 4,
      "tags": Array [
        "quia omnis",
        "et ducimus",
        "neque cum",
        "eveniet magnam",
        "vitae sint",
        "molestiae fugit",
      ],
      "title": "rivulet",
    },
    Object {
      "cover": "https://loremflickr.com/1280/1910/nature",
      "desc": Object {
        "text": "Natus qui eius magni autem iusto. Libero fugit consequatur voluptatibus nisi voluptas enim hic voluptate. Aliquam impedit consectetur. Praesentium libero quisquam. In eligendi labore. Numquam quis voluptas. Aut aut perferendis laborum autem. Sunt et dolore reprehenderit rerum dolorem itaque. In modi minima. Quisquam velit asperiores nemo ipsam iste amet at cumque rerum. Velit est id autem soluta deleniti voluptas. Aut libero nihil iusto deserunt qui id eveniet officia totam.",
      },
      "id": "bk60",
      "mate": Object {
        "author": "Urban",
      },
      "rates": 85,
      "tags": Array [
        "sit dignissimos",
        "voluptas veniam",
      ],
      "title": "ceramic",
    },
    Object {
      "cover": "https://loremflickr.com/1280/1910/abstract",
      "desc": Object {
        "text": "Alias aliquid quidem. Iste ab repellendus sunt suscipit. Nulla odit debitis est. Exercitationem non quia ratione itaque molestias qui libero aspernatur. Corporis in repudiandae aut ipsam soluta. Vel vel facere autem modi quos qui eaque. Ea quibusdam totam dolor officiis et aut est corporis. Voluptas est non fugit occaecati numquam dolor fugit ipsa consequatur. Repellat laboriosam nulla eveniet voluptatem fugiat tempore veritatis vitae cumque. Voluptatum est tempora itaque. Nesciunt unde doloribus est est sed et pariatur. Aut vero impedit quidem pariatur non sequi voluptas. Illo nemo voluptatibus deserunt amet dolorum soluta culpa itaque esse.",
      },
      "id": "bk61",
      "mate": Object {
        "author": "Janet",
      },
      "rates": 74,
      "tags": Array [
        "itaque quos",
        "assumenda et",
        "quia impedit",
        "rerum enim",
      ],
      "title": "downforce",
    },
    Object {
      "cover": "https://loremflickr.com/1280/1910/nightlife",
      "desc": Object {
        "text": "Et tempore accusamus et porro doloribus. Voluptatibus minus voluptatibus omnis quo. Voluptas unde neque sit sit id aut voluptas maxime. Inventore quasi impedit deleniti nobis sed perferendis repellendus. Occaecati illo voluptates perspiciatis sequi a. Sint tenetur quod. Quod eaque ducimus aperiam mollitia. Ullam voluptas ad temporibus fugiat qui. Corrupti amet at quidem cumque facilis. Perspiciatis quo placeat aspernatur quos.",
      },
      "id": "bk92",
      "mate": Object {
        "author": "Bria",
      },
      "rates": 97,
      "tags": Array [
        "quia consequatur",
        "excepturi vel",
        "doloribus tempora",
        "porro illum",
        "est aliquid",
      ],
      "title": "dueling",
    },
    Object {
      "cover": "https://loremflickr.com/1280/1910/nature",
      "desc": Object {
        "text": "Esse autem iusto quia rem maxime reprehenderit autem ea optio. Dolorum autem porro. Aut consequatur pariatur soluta magnam laborum exercitationem eligendi. Dolor corrupti earum rerum non. Vero aperiam animi eum vero aspernatur officiis repellendus commodi. Qui in maxime. Ipsa eaque debitis ab consequatur provident nihil fuga qui. Delectus tenetur sint et et nemo. Deleniti fuga eveniet sit fugiat ea consequatur repellendus. Autem quis fugit est nihil itaque inventore aut. Est placeat delectus veritatis veritatis ipsa voluptatum ut ratione. Aut rerum vel sit blanditiis voluptatum autem occaecati ratione animi. Iure odio delectus veniam neque tenetur natus.",
      },
      "id": "bk92",
      "mate": Object {
        "author": "Marta",
      },
      "rates": 0,
      "tags": Array [
        "sit inventore",
        "aspernatur fugiat",
        "molestiae distinctio",
        "vel soluta",
        "quasi saepe",
        "quo nihil",
        "perspiciatis eveniet",
      ],
      "title": "fortune",
    },
    Object {
      "cover": "https://loremflickr.com/1280/1910/sports",
      "desc": Object {
        "text": "Quod ex cum qui sit qui eum qui exercitationem autem. Quae eligendi cupiditate sed nihil deserunt. Consequuntur aut placeat sequi magni. Voluptatem unde occaecati animi eum molestiae a sunt labore. Corporis quia et sed ullam. Id et officiis aut. Blanditiis eveniet doloribus ea. Autem dolorem tenetur reprehenderit ut excepturi qui. Delectus est nihil quidem hic est. Provident cumque est esse totam voluptas et in dolores. Quo et id ea minima repellendus quia sed porro necessitatibus.",
      },
      "id": "bk57",
      "mate": Object {
        "author": "Florida",
      },
      "rates": 34,
      "tags": Array [
        "beatae natus",
        "sequi accusamus",
      ],
      "title": "mist",
    },
    Object {
      "cover": "https://loremflickr.com/1280/1910/technics",
      "desc": Object {
        "text": "Ut esse nihil aut possimus. Fuga illo rerum et quia molestiae. Non quis numquam alias quos provident quis eum molestiae et. Ullam occaecati omnis hic magni. Autem consequatur reiciendis voluptatibus doloremque ut. Iusto aut tenetur atque sed placeat. Et omnis praesentium sed quod aspernatur tempore. Libero modi dolorem et eum ipsum vel tenetur. Velit veritatis veniam suscipit possimus. Alias corrupti tempore sapiente modi et necessitatibus quos ut. Molestiae alias accusantium atque. Iusto aut minus ad ab inventore repellendus.",
      },
      "id": "bk33",
      "mate": Object {
        "author": "Muhammad",
      },
      "rates": 29,
      "tags": Array [
        "est ea",
        "iste vitae",
        "et eaque",
        "magnam sed",
        "quidem modi",
        "id voluptate",
        "aperiam qui",
      ],
      "title": "tablecloth",
    },
    Object {
      "cover": "https://loremflickr.com/1280/1910/cats",
      "desc": Object {
        "text": "Totam quia voluptatum voluptas. Nobis nulla molestias autem quia consequatur commodi. Qui quia ipsam velit est. Temporibus voluptas et deserunt nemo voluptate et inventore facere. Atque tempora qui ab atque incidunt assumenda similique laborum. Facilis eius quo temporibus sequi omnis quidem. Soluta et error mollitia aut amet ut facere maiores. Dolor voluptas vel et. At molestias debitis est. Veritatis veniam iste.",
      },
      "id": "bk92",
      "mate": Object {
        "author": "Lenora",
      },
      "rates": 48,
      "tags": Array [
        "enim quae",
        "ut voluptates",
        "voluptates ad",
        "vitae explicabo",
        "et repudiandae",
        "aliquam magni",
        "quia suscipit",
        "deserunt perferendis",
      ],
      "title": "sake",
    },
  ],
  "error": undefined,
  "isLoading": false,
  "isValidating": false,
  "mutate": [Function],
}
`;

exports[`useBooksInfinite should return books 1`] = `
Object {
  "data": Array [
    Array [
      Object {
        "cover": "https://loremflickr.com/1280/1910/sports",
        "desc": Object {
          "text": "Quis dolore similique accusantium officiis adipisci. Dolorum cum et totam eveniet culpa qui quibusdam ad facere. Rerum distinctio atque consequatur amet iure sequi voluptas dolorem. Suscipit non quam. Deserunt omnis sint similique cum aliquid est sunt odio nemo. Inventore est ut ut magnam ea dolores ipsum ea recusandae. Et facere cumque quae fuga dolor ipsa dolor. Unde sit cupiditate. Sint aspernatur ullam a est repellendus ducimus. Maxime id est dolor nobis fuga.",
        },
        "id": "bk90",
        "mate": Object {
          "author": "Gracie",
        },
        "rates": 78,
        "tags": Array [
          "quia doloremque",
          "dolorum et",
          "sed ut",
          "facilis et",
          "dolorem non",
        ],
        "title": "catch",
      },
      Object {
        "cover": "https://loremflickr.com/1280/1910/cats",
        "desc": Object {
          "text": "Impedit occaecati magni fuga et cumque excepturi enim voluptas. Quasi ipsum ut. Quia exercitationem sequi voluptates delectus ullam laborum. Porro et consequuntur harum. Hic praesentium maiores qui asperiores qui vero accusantium nemo voluptas. Sequi voluptatum perferendis aperiam qui assumenda velit necessitatibus. Adipisci animi molestias aspernatur quas vel consequatur veniam. Nisi voluptas eos. Facere laboriosam rerum quidem possimus nisi sit totam veniam. Repellendus magni et similique consequatur vel officiis ullam. Est quidem ullam eum voluptas porro et voluptates doloribus. Aut minima reprehenderit commodi nemo ducimus atque. Minima neque natus consequatur pariatur odit quaerat.",
        },
        "id": "bk88",
        "mate": Object {
          "author": "Lonnie",
        },
        "rates": 95,
        "tags": Array [
          "aut laborum",
          "perferendis harum",
          "quo minus",
          "quis consequatur",
          "natus nihil",
          "dicta et",
          "voluptate placeat",
          "et aut",
        ],
        "title": "cheek",
      },
      Object {
        "cover": "https://loremflickr.com/1280/1910/city",
        "desc": Object {
          "text": "Dolor ut quos odio quam nihil rerum quia assumenda. Perspiciatis exercitationem in reprehenderit. Molestiae aut quia voluptatem. Voluptatum dolore officiis facere ea hic nihil voluptatem ut. Consequuntur accusamus similique odit. Repellat culpa explicabo. In ex harum veniam. Corporis fuga saepe facilis. Ut labore similique molestias nihil. Aspernatur reprehenderit neque natus. Consequatur blanditiis odio ad blanditiis ab.",
        },
        "id": "bk41",
        "mate": Object {
          "author": "Raegan",
        },
        "rates": 4,
        "tags": Array [
          "quia omnis",
          "et ducimus",
          "neque cum",
          "eveniet magnam",
          "vitae sint",
          "molestiae fugit",
        ],
        "title": "rivulet",
      },
      Object {
        "cover": "https://loremflickr.com/1280/1910/nature",
        "desc": Object {
          "text": "Natus qui eius magni autem iusto. Libero fugit consequatur voluptatibus nisi voluptas enim hic voluptate. Aliquam impedit consectetur. Praesentium libero quisquam. In eligendi labore. Numquam quis voluptas. Aut aut perferendis laborum autem. Sunt et dolore reprehenderit rerum dolorem itaque. In modi minima. Quisquam velit asperiores nemo ipsam iste amet at cumque rerum. Velit est id autem soluta deleniti voluptas. Aut libero nihil iusto deserunt qui id eveniet officia totam.",
        },
        "id": "bk60",
        "mate": Object {
          "author": "Urban",
        },
        "rates": 85,
        "tags": Array [
          "sit dignissimos",
          "voluptas veniam",
        ],
        "title": "ceramic",
      },
      Object {
        "cover": "https://loremflickr.com/1280/1910/abstract",
        "desc": Object {
          "text": "Alias aliquid quidem. Iste ab repellendus sunt suscipit. Nulla odit debitis est. Exercitationem non quia ratione itaque molestias qui libero aspernatur. Corporis in repudiandae aut ipsam soluta. Vel vel facere autem modi quos qui eaque. Ea quibusdam totam dolor officiis et aut est corporis. Voluptas est non fugit occaecati numquam dolor fugit ipsa consequatur. Repellat laboriosam nulla eveniet voluptatem fugiat tempore veritatis vitae cumque. Voluptatum est tempora itaque. Nesciunt unde doloribus est est sed et pariatur. Aut vero impedit quidem pariatur non sequi voluptas. Illo nemo voluptatibus deserunt amet dolorum soluta culpa itaque esse.",
        },
        "id": "bk61",
        "mate": Object {
          "author": "Janet",
        },
        "rates": 74,
        "tags": Array [
          "itaque quos",
          "assumenda et",
          "quia impedit",
          "rerum enim",
        ],
        "title": "downforce",
      },
      Object {
        "cover": "https://loremflickr.com/1280/1910/nightlife",
        "desc": Object {
          "text": "Et tempore accusamus et porro doloribus. Voluptatibus minus voluptatibus omnis quo. Voluptas unde neque sit sit id aut voluptas maxime. Inventore quasi impedit deleniti nobis sed perferendis repellendus. Occaecati illo voluptates perspiciatis sequi a. Sint tenetur quod. Quod eaque ducimus aperiam mollitia. Ullam voluptas ad temporibus fugiat qui. Corrupti amet at quidem cumque facilis. Perspiciatis quo placeat aspernatur quos.",
        },
        "id": "bk92",
        "mate": Object {
          "author": "Bria",
        },
        "rates": 97,
        "tags": Array [
          "quia consequatur",
          "excepturi vel",
          "doloribus tempora",
          "porro illum",
          "est aliquid",
        ],
        "title": "dueling",
      },
      Object {
        "cover": "https://loremflickr.com/1280/1910/nature",
        "desc": Object {
          "text": "Esse autem iusto quia rem maxime reprehenderit autem ea optio. Dolorum autem porro. Aut consequatur pariatur soluta magnam laborum exercitationem eligendi. Dolor corrupti earum rerum non. Vero aperiam animi eum vero aspernatur officiis repellendus commodi. Qui in maxime. Ipsa eaque debitis ab consequatur provident nihil fuga qui. Delectus tenetur sint et et nemo. Deleniti fuga eveniet sit fugiat ea consequatur repellendus. Autem quis fugit est nihil itaque inventore aut. Est placeat delectus veritatis veritatis ipsa voluptatum ut ratione. Aut rerum vel sit blanditiis voluptatum autem occaecati ratione animi. Iure odio delectus veniam neque tenetur natus.",
        },
        "id": "bk92",
        "mate": Object {
          "author": "Marta",
        },
        "rates": 0,
        "tags": Array [
          "sit inventore",
          "aspernatur fugiat",
          "molestiae distinctio",
          "vel soluta",
          "quasi saepe",
          "quo nihil",
          "perspiciatis eveniet",
        ],
        "title": "fortune",
      },
      Object {
        "cover": "https://loremflickr.com/1280/1910/sports",
        "desc": Object {
          "text": "Quod ex cum qui sit qui eum qui exercitationem autem. Quae eligendi cupiditate sed nihil deserunt. Consequuntur aut placeat sequi magni. Voluptatem unde occaecati animi eum molestiae a sunt labore. Corporis quia et sed ullam. Id et officiis aut. Blanditiis eveniet doloribus ea. Autem dolorem tenetur reprehenderit ut excepturi qui. Delectus est nihil quidem hic est. Provident cumque est esse totam voluptas et in dolores. Quo et id ea minima repellendus quia sed porro necessitatibus.",
        },
        "id": "bk57",
        "mate": Object {
          "author": "Florida",
        },
        "rates": 34,
        "tags": Array [
          "beatae natus",
          "sequi accusamus",
        ],
        "title": "mist",
      },
      Object {
        "cover": "https://loremflickr.com/1280/1910/technics",
        "desc": Object {
          "text": "Ut esse nihil aut possimus. Fuga illo rerum et quia molestiae. Non quis numquam alias quos provident quis eum molestiae et. Ullam occaecati omnis hic magni. Autem consequatur reiciendis voluptatibus doloremque ut. Iusto aut tenetur atque sed placeat. Et omnis praesentium sed quod aspernatur tempore. Libero modi dolorem et eum ipsum vel tenetur. Velit veritatis veniam suscipit possimus. Alias corrupti tempore sapiente modi et necessitatibus quos ut. Molestiae alias accusantium atque. Iusto aut minus ad ab inventore repellendus.",
        },
        "id": "bk33",
        "mate": Object {
          "author": "Muhammad",
        },
        "rates": 29,
        "tags": Array [
          "est ea",
          "iste vitae",
          "et eaque",
          "magnam sed",
          "quidem modi",
          "id voluptate",
          "aperiam qui",
        ],
        "title": "tablecloth",
      },
      Object {
        "cover": "https://loremflickr.com/1280/1910/cats",
        "desc": Object {
          "text": "Totam quia voluptatum voluptas. Nobis nulla molestias autem quia consequatur commodi. Qui quia ipsam velit est. Temporibus voluptas et deserunt nemo voluptate et inventore facere. Atque tempora qui ab atque incidunt assumenda similique laborum. Facilis eius quo temporibus sequi omnis quidem. Soluta et error mollitia aut amet ut facere maiores. Dolor voluptas vel et. At molestias debitis est. Veritatis veniam iste.",
        },
        "id": "bk92",
        "mate": Object {
          "author": "Lenora",
        },
        "rates": 48,
        "tags": Array [
          "enim quae",
          "ut voluptates",
          "voluptates ad",
          "vitae explicabo",
          "et repudiandae",
          "aliquam magni",
          "quia suscipit",
          "deserunt perferendis",
        ],
        "title": "sake",
      },
    ],
  ],
  "error": undefined,
  "isLoading": false,
  "isValidating": false,
  "mutate": [Function],
  "setSize": [Function],
  "size": 1,
}
`;
